### PR TITLE
Fix permutation copy

### LINKED ISSUE #25378: Permutation is reconstructed wrongly from pickle
# Description

Pickling a `sympy.combinatorics.Permutation` instance and loading the pickle again wrongly reconstructs the object.

Reproduction (in isympy session):
```
In [1]: from sympy.combinatorics import Permutation

In [2]: a = Permutation([2,1,0])

In [3]: import pickle

In [4]: with open("test.pickle", 'wb') as f:
   ...:     pickle.dump(a, f)
   ...: 

In [5]: with open("test.pickle", 'rb') as f:
   ...:     b = pickle.load(f)
   ...: 

In [6]: a
Out[6]: Permutation(0, 2)

In [7]: b
Out[7]: Permutation()

```

# Software versions
Sympy version: 1.12
Python version: 3.11.3
OS: Arch Linux

Also reproducible on <https://live.sympy.org/>

# Additional information
Note that this causes `sympy.tensor.tensor.TensorSymmetry` to be wrongly reconstructed from pickles, and thus no tensorial expression can be pickled. 

It seems this may be related to https://github.com/sympy/sympy/issues/22310, but I don't understand why this is only an issue for `Permutation`, and not, say, `Symbol`.

### PR DESCRIPTION
**Pull request by idanpa**
#### References to other Issues or PRs
Fixes #25378

#### Brief description of what is fixed or changed
The _args attribute of sympy.core.basic.Basic was not set correctly, then copying permutation (e.g. pickle) would fail when calling __new__ with empty _args.

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/core/traversal.py [hunk 1]
```diff
@@ -29,8 +29,8 @@ def iterargs(expr):
         yield i
         try:
             args.extend(i.args)
-        except TypeError:
-            pass  # for cases like f being an arg
+        except AttributeError:
```
Comment (smichr) [inline sympy/core/traversal.py:32]: Do you have a test that can test this?

Comment (idanpa) [reply sympy/core/traversal.py:32]: Thanks for the review @smichr !

Actually - couldn't find an example beside the one I introduced in the next commit (where args of permutation is its array form with numbers).
This made me realize I better stick to the way other children of Basic are solving this problem, by reimplementing __getnewargs__ (like Numbers, Str, etc.)
So, I changed the commit above, and now this change is not needed for this pull-request anymore.
(I added it so CI would pass)

The options I see:
1. Keep this change without testing it (the original purpose here was AttributeError, right?)
2. Discard this change from this pull-request
3. Remove this error handling completely (seems to me it is not needed, although I'm not sure).
4. Add a test that implements a custom child of Basic with non-Basic args that would make this error handling useful.

Please let me know what you want me to do.

**Comment by smichr** [comment]
All arts of sympy objects should be Basic. I would go with opt 3. Just add another commit that removes that.

**Comment by idanpa** [comment]
> All arts of sympy objects should be Basic. I would go with opt 3. Just add another commit that removes that.

Done.
Please review.

File: sympy/combinatorics/permutations.py [hunk 2]
```diff
@@ -1050,6 +1050,12 @@ def _af_new(cls, perm):
         p._size = len(perm)
         return p
 
+    def copy(self):
+        return self.__class__(self.array_form)
+
+    def __getnewargs__(self):
+        return (self.array_form,)
```
Comment (sylee957) [inline sympy/combinatorics/permutations.py:1057]: How about using `_hashable_content`

**Comment by sylee957** [comment]
Can there be any test for original pickling issue?
I'm not obvious if the logic here eventually connects to the E2E tests about pickle.

File: sympy/combinatorics/permutations.py (hunk 2 above)
Comment (idanpa) [reply sympy/combinatorics/permutations.py:1057]: For me the name `_hashable_content` doesn't suggest the entire object can be reconstructed from it.
I see `array_form` used extensively throughout the code to represent this object (and no usage of `_hashable_content`).
WDYS?

**Comment by idanpa** [comment]
> Can there be any test for original pickling issue? I'm not obvious if the logic here eventually connects to the E2E tests about pickle.

The underlying problem of the pickling issue, is that python doesn't reconstruct the object correctly - passing empty args to `__new__`.
A simpler code that fails right now is this one:
``` python
from copy import copy
from sympy.combinatorics import Permutation
a = Permutation([[2,1,0],[6,7]])
assert a == copy(a)
```
We can test the entire pickling flow from #25378, but for me it seems like the `copy` flow is good enough.
The [copy docs](https://docs.python.org/3/library/copy.html) also says ("...is compatible with the way these are treated by the pickle module")

**Pull request merged**